src/app/token/actions.js

```diff
@@ -31,6 +31,8 @@ export const fetch = (ticker: string) => async (dispatch: Function, getState: Ge
     expires.setDate(expires.getDate() + 1)
     const token: SecurityToken = await SecurityTokenRegistry.getTokenByTicker(ticker)
     if (token && token.owner !== getState().network.account) {
+      // $FlowFixMe
+      getState().pui.common.history.push('/')
```

Comment:
I just rechecked task description and it says that "404 pages should be redirecting users to the main page". This code will redirect only when issuer tries to open token that does not belong to him. Moreover, I am absolutely against such strategy. Reassigned task back to Josep and added comments there